docs/internal/how-to-update-the-build-image.md

```diff
@@ -1,8 +1,14 @@
-The build image currently can only be updated by a Grafana Mimir maintainer. If you're not a maintainer you can still open a PR with the changes, asking a maintainer to assist you in publishing the updated image. The procedure is:
+The build image currently can only be updated by a Grafana Mimir maintainer. If you're not a maintainer you can still open a PR with the changes, asking a maintainer to assist you in publishing the updated image. The manual procedure is:
 
 1. Update `mimir-build-image/Dockerfile` on a new branch. Note: the resulting images have the tag name derived from the branch name.
 2. Make sure to have [Docker Buildx](https://docs.docker.com/buildx/working-with-buildx/). Docker Desktop and major distributions have it in the docker package.
 3. On Linux you'll need some extra qemu packages as well: `sudo apt-get install qemu qemu-user-static binfmt-support debootstrap`. And set up docker buildx: `docker buildx create --name armBuilder ; docker buildx use armBuilder`.
 4. Build and publish the image by using `make push-multiarch-build-image`. This will build and push multiplatform docker image (for Linux/amd64 and Linux/arm64).
 5. Replace the image tag in Makefile (variable `LATEST_BUILD_IMAGE_TAG`).
 6. Open a PR and make sure the CI with the new build image passes
+
+The automatic procedure is:
+1. Update `mimir-build-image/Dockerfile` and create a pull request.
+2. Request a maintainer to trigger the build of mimir-build-image by commenting "/build-mimir-image" on the pull request.
```

Comment:
Protection like this likely won't work as anyone who wants to do something malicious could modify the workflow to remove this requirement as part of their PR. 

I think we instead need to use GitHub's [workflow approval option for outside collaborators](https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks). We currently have this set to "require approval for first-time contributors", but should probably change this to "require approval for all outside collaborators".

Reply:
agreed, it would simplify also the workflow itself.

Reply:
Repository has been configured to require approval for all runs of GHA on PRs from outside collaborators.